packages/_helpers/decorators.ts

```diff
@@ -0,0 +1,207 @@
+          // tslint:disable-next-line:no-invalid-this
+          value._configByDecorator( this, eventName as string );
+          propertyValue = value;
+        } else {
+          throw new Error( `${propertyKey} must be an instance of EventEmiiter` );
+        }
+      }
+    };
+  };
+}
+
+export class EventEmitter<T> {
+
+  private origin: Component<any>;
+  private eventName: string;
+
+  constructor(
+    private eventConfig: Object = {}
+  ) { }
+
+  emit( data: T ): boolean {
+    return emit(
+      this.origin,
+      this.eventName,
+      {
+        ...this.eventConfig,
+        ...{ detail: data }
+      }
+    );
+  }
+
+  _configByDecorator( origin: Component<any>, eventName: string ) {
+    this.origin = origin;
+    this.eventName = eventName;
+  }
+}
+
+function parseType( type: Function ): PropType {
+  if ( typeof type !== 'function' ) {
+    return;
+  }
+  const inst = type() as Array<any> | Object | number | boolean | string;
+  if ( inst instanceof Array ) {
+    return 'array';
+  }
+  if ( typeof inst === 'object' ) {
+    return 'object';
+  }
+  return ( typeof inst ) as 'boolean' | 'number' | 'string';
+}
+
+export function colored(): ClassDecorator {
+  return function <T extends typeof Component>( Target: T ): T {
+
+    const newProps = {
+      ...( Target as any ).props,
+      ...{ color: skProp.string<any, ColorType>( { attribute: { source: true } } ) }
+    };
+    Object.defineProperty( Target, 'props', {
+      configurable: true,
+      get() { return newProps; }
+    } );
+    return Target;
+  };
+}
+
+export function disabled(): ClassDecorator {
+  return function <T extends typeof Component>( Target: T ): T {
+
+    const newProps = {
+      ...Target.props,
+      ...{ disabled: skProp.boolean( { attribute: true } ) }
+    };
+    Object.defineProperty( Target, 'props', {
+      configurable: true,
+      get() { return newProps; }
+    }
+    );
+    return Target;
+  };
+}
+
+
+export function shadyCssStyles() {
+  return function <T extends Constructable<Component<any>>>( Target: T ) {
+    const proto = Target.prototype;
+    const originalRenderCallback = proto.renderCallback;
+
+    Object.defineProperties( proto, {
+      shadyCss: {
+        get( this: { css: string } ) {
+          // tslint:disable-next-line:no-invalid-this
+          if ( IS_DEV && !( 'css' in this ) ) {
+            throw new Error( `you have to implement 'css' property when using '@shadyCssStyles' Class Decorator!` );
+          }
+          // tslint:disable-next-line:no-invalid-this
+          return scopeCss( this as any, this.css );
+        }
+      },
+      renderCallback: {
+        value( this: { shadyCss: string }, ...args: any[] ) {
+          // tslint:disable-next-line:no-invalid-this
+          return [ h( 'style', this.shadyCss ) ].concat( originalRenderCallback.apply( this, args ) );
+        }
+      }
+    } );
+
+    return Target;
+
+    // return class extends Target {
```

Comment:
Yup. I have to rebase and format and cleanup after team review) discussion. I don't like to add one reviewer for big feature ( like with app layout for instance ehm )